cuda/solver/gmres_kernels.cu

```diff
@@ -688,17 +688,11 @@ void solve_upper_triangular(
 
 
 template <typename ValueType>
-void solve_x(std::shared_ptr<const CudaExecutor> exec,
-             const matrix::Dense<ValueType> *krylov_bases,
+void solve_x(const matrix::Dense<ValueType> *krylov_bases,
              const matrix::Dense<ValueType> *y, matrix::Dense<ValueType> *x,
-             const Array<size_type> *final_iter_nums,
-             const LinOp *preconditioner)
+             matrix::Dense<ValueType> *before_preconditioner,
```

Comment:
I think you can remove the `x` parameter from here as you don't need it inside the function anymore.